app/src/main/java/org/wikipedia/main/MainActivity.kt

```diff
@@ -83,12 +81,16 @@ class MainActivity : SingleFragmentActivity<MainFragment>(), MainFragment.Callba
         override fun onReceive(context: Context?, intent: Intent?) {
             if (intent != null && context != null) {
                 if (intent.action == ENVOY_BROADCAST_VALIDATION_SUCCEEDED) {
-                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED)
-                    val validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED)
+                    val validUrl = intent.getStringExtra(ENVOY_DATA_URL_SUCCEEDED) ?: ""
+                    var validService = intent.getStringExtra(ENVOY_DATA_SERVICE_SUCCEEDED) ?: ""
+                    if (validUrl.startsWith("envoy") && validService.startsWith("http")) {
+                        validService = "envoy"
+                    }
+                    val sanitizedUrl = UrlUtil.sanitizeUrl(validUrl, validService)
```

Comment:
is sanitizing related to snowflake changes somehow?

Reply:
i may have overemphasized the snowflake aspect.  this pr is more about both updating the secrets (which includes adding snowflake) and obfuscating them better before the next release